slither/solc_parsing/declarations/function.py

```diff
@@ -1149,16 +1149,16 @@ def _find_end_loop(self, node: Node, visited: List[Node], counter: int) -> Optio
 
         return None
 
-    def _find_start_loop(self, node: Node, visited: List[Node]) -> Optional[Node]:
+    def _find_if_loop(self, node: Node, visited: List[Node]) -> Optional[Node]:
         if node in visited:
             return None
 
-        if node.type == NodeType.STARTLOOP:
+        if node.type == NodeType.IFLOOP:
```

Comment:
Can you change it to this code which should work correctly for while and for. For while returns the IFLOOP while for the for returns the increment expression. The assumption is that a while always have BEGINLOOP and ENDIF as fathers. I already did some tests but please test this with all the edge cases you can think of.
```
        if node.type == NodeType.IFLOOP:
            for father in node.fathers:
                if father.type == NodeType.EXPRESSION:
                    return father
            return node
```

Reply:
Could you write how `_find_if_loop` method should look like after those changes, I cannot figure out where to put this code?

Reply:
```  
def _find_if_loop(self, node: Node, visited: List[Node], skip_if_loop: int) -> Optional[Node]:
        if node in visited:
            return None

        # Code from above

        visited = visited + [node]
        for father in node.fathers:
            ret = self._find_if_loop(father, visited, skip_if_loop)
            if ret:
                return ret

        return None
```
You also need to add a 0 as third argument [here](https://github.com/crytic/slither/blob/709d5e925fc363fda33951e6aa035f0d9366f9cf/slither/solc_parsing/declarations/function.py#L1184)

Reply:
I couldn't find any problems with the modifications you had provided. I've committed the changes, also I couldn't find a way to update integration tests